src/ILCompiler/src/Program.cs

```diff
@@ -67,61 +68,37 @@ private void InitializeDefaultOptions()
+            IReadOnlyList<string> implicitInputFiles = Array.Empty<string>();
+            IReadOnlyList<string> referenceFiles = Array.Empty<string>();
 
-            string option;
-            while ((option = parser.GetOption()) != null)
+            AssemblyName name = typeof(Program).GetTypeInfo().Assembly.GetName();
+            ArgumentSyntax.Parse(args, syntax =>
             {
-                switch (option.ToLowerInvariant())
-                {
-                    case "?":
-                    case "help":
-                        _help = true;
-                        break;
-
-                    case "":
-                    case "in":
-                        parser.AppendExpandedPaths(_inputFilePaths, true);
-                        break;
-
-                    case "o":
-                    case "out":
-                        _options.OutputFilePath = parser.GetStringValue();
-                        break;
-
-                    case "dgmllog":
-                        _options.DgmlLog = parser.GetStringValue();
-                        break;
-
-                    case "fulllog":
-                        _options.FullLog = true;
-                        break;
-
-                    case "verbose":
-                        _options.Verbose = true;
-                        break;
-
-                    case "r":
-                    case "reference":
-                        parser.AppendExpandedPaths(_referenceFilePaths, false);
-                        break;
-
-                    case "cpp":
-                        _options.IsCppCodeGen = true;
-                        break;
-
-                    case "nolinenumbers":
-                        _options.NoLineNumbers = true;
-                        break;
-
-                    case "systemmodule":
-                        _options.SystemModuleName = parser.GetStringValue();
-                        break;
-
-                    default:
-                        throw new CommandLineException("Unrecognized option: " + parser.GetCurrentOption());
-                }
-            }
+                syntax.ApplicationName = name.Name + " " + name.Version.ToString();
+
+                syntax.HandleHelp = false;
+                syntax.HandleErrors = false;
+            
+                syntax.DefineOption("h|help", ref _help, "Help message for ILC");
+                syntax.DefineOptionList("i|in", ref explicitInputFiles, "Input file(s) to compile");
+                syntax.DefineOptionList("r|reference", ref referenceFiles, "Reference file(s) for compilation");
+                syntax.DefineOption("o|out", ref _options.OutputFilePath, "Output file for compilation");
+                syntax.DefineOption("cpp", ref _options.IsCppCodeGen, "Compile for C++ code-generation");
+                syntax.DefineOption("nolinenumbers", ref _options.NoLineNumbers, "Debug line numbers for C++ code-generation");
```

Comment:
This option is currently hooked up to both ryujit and cppcodegen, and it also controls pretty printing for cppcodegen. Could you please open issue to get this sorted out separately?